### PR TITLE
Cleanup swapped Image model

### PR DESCRIPTION
**Pull request by yakky**
This provides a bit of cleanup. 

Polymorphic issues with django-polymorphic<2.0 are still there, but other issues with the current implementation have been fixed

Partial fix for #1025

### CODE REVIEW

**Comment by yakky** [comment]
@skirsdeda what do  you think about this?

**Comment by yakky** [comment]
@czpython any opinion from your side?

**Comment by yakky** [comment]
Unfortunately the only way to avoid the polymorphic bug is to skip the default image entirely, otherwise it's discovered (because it's imported in `models.py`) and polymorphic complains about it
As the polymorphic issue is triggered in Django 1.10 only, we can avoid importing the default image in 1.10, but this does not make the solution less ugly

File: filer/models/imagemodels.py [hunk 1]
```diff
@@ -11,43 +11,45 @@
...
-                    else:
-                        self.date_taken = datetime(
-                            int(year), int(month), int(day),
-                            int(hour), int(minute), int(second))
-            except Exception:
-                pass
-        if self.date_taken is None:
-            self.date_taken = now()
-        super(Image, self).save(*args, **kwargs)
+if filer_settings.FILER_IMAGE_MODEL == 'filer.Image':
```
[the remark this answers is not in the document]
Comment (yakky) [reply filer/models/imagemodels.py]: I know, it's terrible, but see https://github.com/divio/django-filer/pull/1063#issuecomment-381351198

**Comment by yakky** [comment]
@skirsdeda what would you think about marking the `filer.Image` model as abstract if it's swapped?
It seems to work well, and it looks less ugly to me

**Comment by skirsdeda** [comment]
@yakky seeing how simple the fix in django-polymorphic is, I think it would be better to convince @vdboor to do an extra release. Marking ```filer.Image``` as abstract when it's swapped would also introduce migration problems, I guess. Also, with normal setup it should be possible to switch to swapped model in existing project with previous migrations. Not sure if that would work either.

**Comment by yakky** [comment]
@skirsdeda I'll try to reach him by email.

**Comment by yakky** [comment]
Merging this and releasing 1.3.1 as the current situation in develop is better than 1.3
We will see if this can be fixed with a new polymorphic release

**Comment by vdboor** [comment]
@yakky I just noticed this thread, but didn't see anything in django-polymorphic. Is there a commit or issue I should look at?

**Comment by skirsdeda** [comment]
@vdboor see https://github.com/django-polymorphic/django-polymorphic/issues/270

**Comment by vdboor** [comment]
@yakky @skirsdeda thanks! I've released django-polymorphic v1.3.1. Since cherry-picking was so simply, I've also cherry-picked almost every bugfix from 2.0 into 1.3.1. The major remaining difference between both versions is the Django 2.0 support.

**Comment by yakky** [comment]
@vdboor awesome :bowing_man:  ! And Thanks @skirsdeda for your efforts to fix this!

File: filer/utils/compatibility.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ def truncate_words(s, num, end_text='...'):
 LTE_DJANGO_1_8 = django.VERSION < (1, 9)
 LTE_DJANGO_1_9 = django.VERSION < (1, 10)
 GTE_DJANGO_1_10 = django.VERSION >= (1, 10)
+DJANGO_1_10 = django.VERSION[:2] == (1, 10)
```
Comment (czpython) [inline filer/utils/compatibility.py]: @yakky this doesn't seem to be used at all?

**Pull request merged**